docs/source/concepts/local-ledger.rst

```diff
@@ -0,0 +1,296 @@
+   
+   The rationale for this behaviour is that Alice could have learnt about the contract ID out of band or made it up.
+   The Participant Nodes therefore cannot know whether there will ever be a **Create** event for the contract.
+   So if Participant Nodes delayed outputting the **Create** action for the `CounterOffer` until a **Create** event for the `Iou` contract was published,
+   this delay might last forever and liveness is lost.
+   Causality therefore does not capture data flow through applications.
+   
+#. *Divulged actions do not induce causal order.*
+
+   The painter witnesses the fetching of Alice's `Iou` when the `ShowIou` contract is archived.
+   The painter also witnesses archival of the `Iou` when Alice exercises the transfer choice as a consequence of the painter accepting the `CounterOffer`.
+   However, the *use ordering* rule does not apply as the painter is not a stakeholder of the `Iou`.
+   
+   Consider a setup where two Participant Nodes `N`:sub:`1` and `N`:sub:`2` host the painter.
+   He sees the divulged `Iou` and the created `CounterOffer` through `N`:sub:`1`\ 's transaction tree stream
+   and then submits the acceptance through `N`:sub:`1`.
+   As in the previous example, `N`:sub:`2` does not know about the dependence of the two commits.
+   Therefore, the painter's local ledgers on `N`:sub:`2` need not order the two transactions.
+   Accordingly, `N`:sub:`2` may output the accepting transaction *before* the `ShowIou` contract on the transaction stream.
+      
+   Even though this may seem unexpected, it is in line with stakeholder-based ledgers:
+   Since the painter is not a stakeholder of the `Iou` contract, he should not care about the archivals or creates of the contract.
+   In fact, the divulged `Iou` contract does not show up in the painter's active contract service or in the flat transaction stream.
+   Such witnessed events are included in the transaction tree stream as a convenience:
+   They relieve the painter from computing the consequences of the choice and enable him to check that the action conforms to the model.
+
+   By a similar argument, being an actor of an **Exercise** action induces causal order with respect to other uses of the contract only if the actor is a contract stakeholder.
+   This is because non-stakeholder actors of an **Exercise** action authorize the action, but they have no say in whether the contract is active; this is the signatories' job.
+   
+#. *Causality depends on the party.*
+
+   By the previous example, for the painter, fetching the `Iou` is not causally ordered before transferring the `Iou`.
+   For Alice, however, the **Fetch** is causally ordered before the **Exercise** by the *use ordering* rule
+   because Alice is a stakeholder on the `Iou` contract.
+   This shows that causal ordering depends on the party.
+   Even if both Alice and the painter are hosted on the same Participant Node,
```

Comment:
AFAIK this is not possible in any of our current implementations. Perhaps Corda? Why would we want/need to weaken this guarantee?